src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -901,43 +1014,76 @@ public void reduceNeededResources(final Block block)
         }
     }
 
-    private void spawnEntity(@Nullable final Entity entity)
+    private Boolean spawnEntity(@NotNull final Structure.StructureBlock currentBlock)
     {
-        if (entity != null && job instanceof JobBuilder)
+        final Template.EntityInfo entityInfo = currentBlock.entity;
+        if (entityInfo != null && job instanceof JobBuilder)
         {
-            final BlockPos pos = ((JobBuilder) job).getStructure().getOffsetPosition();
-
-            if (entity instanceof EntityHanging)
+            Entity entity;
+            try
             {
-                @NotNull final EntityHanging entityHanging = (EntityHanging) entity;
-
-                entityHanging.posX += pos.getX();
-                entityHanging.posY += pos.getY();
-                entityHanging.posZ += pos.getZ();
-                //also sets position based on tile
-                entityHanging.setPosition(
-                        entityHanging.getHangingPosition().getX(),
-                        entityHanging.getHangingPosition().getY(),
-                        entityHanging.getHangingPosition().getZ());
+                entity = EntityList.createEntityFromNBT(entityInfo.entityData, world);
+            }
+            catch (Exception e)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Catch a list of specific exception subtypes instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2221)